Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -155,14 +155,32 @@ template <class Type> class TableColumn : public API::Column {
    * is throw. In case of an overflow boost::numeric::positive_overflow or
    * boost::numeric::negative_overflow
    * is throw.
-   * @param i :: The index to an element.
+   * @param value :: The value of the element.
    */
-  double toDouble(size_t i) const override {
+  template <typename T> double convertToDouble(const T &value) const {
     typedef
```

Comment:
I think we should get rid of this `mpl::if` completely and just have `return numeric_cast<double, Type>(value)`. This way we will get a compile-time error straight from the `numeric_cast` if the Type is not convertible to a `double`. There is a boost mpl header that will also need to go.